foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+            raise ValueError(error_str.format(uuid))
+
+    def add_point(self, point):
+        message = 'Point addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_edge(self, edge):
+        message = 'Edge addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_face(self, face):
+        message = 'Face addition not supported yet'
+        raise NotImplementedError(message)
+
+    def add_cell(self, cell):
+        message = 'Cell addition not supported yet'
+        raise NotImplementedError(message)
+
+    def update_point(self, point):
+        """ Updates the information of a point.
+
+        Gets the mesh point identified by the same
+        id as the provided point and updates its information
+        with the one provided with the new point.
+
+        Parameters
+        ----------
+        point : Point
+            Point to be updated
+
+        Raises
+        ------
+        KeyError
+            If the point was not found in the mesh
+
+        TypeError
+            If the object provided is not a point
+
+        """
+
+        if point.uid not in self._uuidToFoamLabel:
+            error_str = "Trying to update a non-existing point with uuid: "\
+                + str(point.uid)
+            raise KeyError(error_str)
+
+        if not isinstance(point, Point):
+            error_str = "Trying to update an object with the wrong type. "\
+                + "Point expected."
+            raise TypeError(error_str)
+
+        foamface.setPointCoordinates(self.name,
+                                     self._uuidToFoamLabel[point.uid],
+                                     point.coordinates)
+
+    def update_edge(self, edge):
+        message = 'Edge update not supported yet'
+        raise NotImplementedError(message)
+
+    def update_face(self, face):
+        """ Updates the information of a face.
+
+        Gets the mesh face identified by the same
+        uuid as the provided face and updates its information
+        with the one provided with the new face.
+
+        Parameters
+        ----------
+        face : Face
+            Face to be updated
+
+        Raises
+        ------
+        KeyError
+            If the face was not found in the mesh
+
+        TypeError
+            If the object provided is not a face
+
+        """
+
+        if face.uid not in self._uuidToFoamLabel:
+            error_str = "Trying to update a non-existing face with uuid: "\
+                + str(face.uid)
+            raise KeyError(error_str)
+
+        if not isinstance(face, Face):
+            error_str = "Trying to update an object with the wrong type. "\
+                + "Face expected."
+            raise TypeError(error_str)
+
+        foamface.setFacePoints(self.name,
+                               self._uuidToFoamLabel[face.uid],
+                               face.points)
+
+    def create_dummy_cellvectordata(self, data_name, dimensionset):
```

Comment:
API suggestion: These methods (create_xxx..) could also be defined as utility functions to avoid confusion since they are not part of the official api.